core/src/test/java/org/elasticsearch/cluster/service/ClusterServiceTests.java

```diff
@@ -382,21 +408,54 @@ public void execute() {
                     counter.incrementAndGet();
                 }
             }
+
+            @Override
+            public boolean equals(Object o) {
+                if (this == o) {
+                    return true;
+                }
+                if (o == null || getClass() != o.getClass()) {
+                    return false;
+                }
+                Task task = (Task) o;
+                return id == task.id;
+
+            }
+
+            @Override
+            public int hashCode() {
+                return id;
+            }
+
+            @Override
+            public String toString() {
+                return Integer.toString(id);
+            }
         }
 
-        int numberOfThreads = randomIntBetween(2, 8);
-        int tasksSubmittedPerThread = randomIntBetween(1, 1024);
+        int numberOfThreads = randomIntBetween(2, 5);
+        int taskSubmissionsPerThread = randomIntBetween(1, 24);
```

Comment:
I think you misunderstood, I am not referring to the original settings, just keeping the number of threads high but accepting the reduction in the number of submissions per thread. You're right that the max is what matters, but I saw a max of less than 20 ms.